Source/WebCore/style/ElementRuleCollector.cpp

```diff
@@ -673,7 +673,7 @@ void ElementRuleCollector::addMatchedProperties(MatchedProperties&& matchedPrope
 
             // The value currentColor has implicitely the same side effect. It depends on the value of color,
             // which is an inherited value, making the non-inherited property implicitly inherited.
-            if (is<CSSPrimitiveValue>(value) && downcast<CSSPrimitiveValue>(value).valueID() == CSSValueCurrentcolor)
+            if (is<CSSPrimitiveValue>(value) && StyleColor::isCurrentColor(downcast<CSSPrimitiveValue>(value)))
```

Comment:
This might be hot code (not 100% sure but I guess @anttijk would know). If so, it might be unfortunate to now have an extra function call and an extra `isValueid()` check.

Reply:
Maybe we can drop the `isValueID()` check like Sam suggested and inline the function in the header to avoid any perf regression risk?

Reply:
Done 👍